e2e/wait/wait.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2023 Sauce Labs Inc. All rights reserved.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package wait
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+type Waiter struct {
+	Client http.Client
+	// Endpoint is the endpoint to check for readiness.
+	Endpoint string
+	// MaxWait is the maximum time to wait for a server to be ready.
+	MaxWait time.Duration
+	// Backoff is the time to wait between checks.
+	Backoff time.Duration
+}
+
+var DefaultWaiter = Waiter{
```

Comment:
Do we need the `DefaultWaiter` to be exported now? How about we make it a function. 

We will use it to make a local variable `defaultWaiter` in the package for `ServerReady` function. 
Users who would like to change the waiter config could call this function and modify the struct.

Reply:
I guess it does not matter. This way it's possible to change the defaults.
I can unexpert it no problem.